heron/api/src/java/com/twitter/heron/api/state/State.java

```diff
@@ -0,0 +1,26 @@
+// Copyright 2016 Twitter. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.twitter.heron.api.state;
+
+import java.io.Serializable;
+import java.util.Map;
+
+/**
+ * State represents the state interface as seen by stateful bolts and spouts.
+ * In Heron, state gives a notional Key/Value interface along with the
+ * ability to iterate over the key/values
+ */
+public interface State extends Map<Serializable, Serializable> {
```

Comment:
@billonahill 
> Can we use generics in the State API for better type safety? (Map<<K extends Serializable>, <V extends Serializable>>)

Out of curiosity, what is the meaning of "type safety" in this sentence? In what way could user implement the `State` interface so that certain "type safety" could be lost?